angular/src/app/viewer/visualization/display-canvas/display-canvas.component.ts

```diff
@@ -1,78 +1,158 @@
-/* eslint-disable @typescript-eslint/no-unsafe-assignment */
-/* eslint-disable @typescript-eslint/no-unsafe-call */
-/* eslint-disable @typescript-eslint/no-unsafe-member-access */
-import { Component, Input, OnInit } from '@angular/core';
-import { Network } from '../network-display/network';
-import * as cytoscape from 'cytoscape';
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  Input,
+  OnInit,
+  ViewChild,
+} from '@angular/core';
+import { Network } from 'src/app/shared/models/backend-types.model';
+import cytoscape, {
+  EdgeDataDefinition,
+  NodeDataDefinition,
+  Stylesheet,
+  EventObjectNode,
+} from 'cytoscape';
+
+import popper from 'cytoscape-popper';
+cytoscape.use(popper);
+
+/** default colours for nodes in the graph */
+const nodeColor = Object.freeze({
+  mRNA: '#007f7f',
+  microRNA: '#827f00',
+  HIGHLIGHT: '#ffde4c',
+  SELECTED: '#00de4c',
+  CONNECTED: '#6fde95',
+});
+
+/** list of shapes that can be used to draw a node */
+const nodeShape = Object.freeze({
+  mRNA: 'ellipse',
+  microRNA: 'pentagon',
+});
 
 @Component({
   selector: 'app-display-canvas',
   templateUrl: './display-canvas.component.html',
   styleUrls: ['./display-canvas.component.scss'],
 })
-export class DisplayCanvasComponent implements OnInit {
-  @Input() network: Network | undefined;
+export class DisplayCanvasComponent implements OnInit, AfterViewInit {
+  private _fetching!: boolean;
```

Comment:
@RodolfoAllendes 
Thanks for all the fixes! I think there might just be this one left:
It turns out I was lying here because the setter for `fetching` has a side effect, but I think `colspan` and `isEmpty` could be simple public properties?

Reply:
@yujiberra 
I added the functionality on the setter for fetching on one of the latest commits...  so it was like that before

The colspan is meant to be similar in the future, specially when dealing with the theoretical display of multiple networks... but yeah, in the meantime it could simply be a public property

and the isEmpty is definetely ok being public

Reply:
@RodolfoAllendes 
> The colspan is meant to be similar in the future, specially when dealing with the theoretical display of multiple networks... but yeah, in the meantime it could simply be a public property

Ah gotcha, in that case I have no problem with keeping colspan as is for now!